CREATE_A_ROLE.md

```diff
@@ -1,32 +1,107 @@
 A basic guide for how to create a new role in oasis-roles
 
-Setup environment
-=================
-
-The following Python tools should be installed in either a virtualenv or in your system or elsewise
-should be on your system path:
-
-* ansible
-* ansible-lint
-* yamllint
-* flake8
-* molecule
-
-Create Role
-===========
-
-* `mkdir -p oasis/roles && cd oasis/roles`
-* `git clone git@github.com:oasis-roles/meta_skeleton.git`
-* `ansible-galaxy init --role-skeleton=meta_skeleton my_role_name`
-* `cd my_role_name`
-* Edit README.md, meta/main.yml, and write your code
-
-Setup CI Services
-=================
-
-* Create repo in GitHub
-* Create Travis CI job for new repo
-* Add GH remote and push code
-* `ansible-galaxy import oasis-roles my_role_name` - adds repo to global Galaxy index
-* `ansible-galaxy setup travis oasis-roles my_role_name [Travis CI API key]` - adds auto-updating Galaxy index
-  when Travis CI jobs are completed
+# Setup environment #
+
+* Install [podman](https://podman.io/) using your system package manager.  This
+  example is for Fedora:
+
+  `sudo dnf install podman-docker`
+
+* Create a virtualenv.
+  [virtualenvwrapper](https://virtualenvwrapper.readthedocs.io/en/latest/install.html#basic-installation)
+  is recommended.
+
+  `mkvirtualenv oasis`
+
+* Install Python dependencies.
+
+  `pip3 install ansible tox-ansible pytest-molecule molecule-openstack molecule-podman molecule-docker os-cloud-config`
+
+* Install the OASIS
+  [ansible_collection_system](https://github.com/oasis-roles/ansible_collection_system)
+  collection via Galaxy to use the
+  [molecule_openstack_ci](https://github.com/oasis-roles/ansible_collection_system/tree/master/roles/molecule_openstack_ci)
+  role for OpenStack provisioning.
+
+  `ansible-galaxy collection install git+https://github.com/oasis-roles/ansible_collection_system.git`
+
+* Clone an existing collection.  In this example we'll be using the
+  [ansible_collection_system](https://github.com/oasis-roles/ansible_collection_system)
+  collection.  Git submodules are used to bring in shared repo dependencies.
+
+  `git clone --recurse-submodules git@github.com:oasis-roles/ansible_collection_system.git`
+
+* Clone the OASIS [meta_skeleton](https://github.com/oasis-roles/meta_skeleton)
+  role template.  We delete the `.git` directory of this repo due to Ansible
+  bug [#71977](https://github.com/ansible/ansible/issues/71977).
+
+  `git clone git@github.com:oasis-roles/meta_skeleton.git && rm -rfv ./meta_skeleton/.git`
+
+
+* To use Subscription Manager via the OASIS
+  [rhsm](https://github.com/oasis-roles/ansible_collection_system/tree/master/roles/rhsm)
+  role, add the following variables to your environment and change the values
+  according to your Subscription Manager configuration.
+
+  ```
+  export OASIS_RHSM_USERNAME=myusername
+  export OASIS_RHSM_PASSWORD=mypassword
+  export OASIS_RHSM_SERVER_HOSTNAME=subscription.rhn.stage.redhat.com
+  export OASIS_RHSM_POOL_IDS=mypoolid
```

Comment:
This variable needs to be an encoded JSON array of strings of the ids. So something like

`export OASIS_RHSM_POOL_IDS=['mypoolid']`

At least, that's how I have it configured on my system.